analysis/make-epochs.py

```diff
@@ -0,0 +1,106 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.11.4
+#   kernelspec:
+#     display_name: tutorialAlice
+#     language: python
+#     name: tutorialalice
+# ---
+
+# +
+from pathlib import Path
+import re
+
+import eelbrain
+import mne
+import trftools
+# -
+
+STIMULI = [str(i) for i in range(1, 13)]
+DATA_ROOT = Path("~").expanduser() / 'Data' / 'Alice'
+PREDICTOR_DIR = DATA_ROOT / 'predictors'
+EEG_DIR = DATA_ROOT / 'eeg'
+SUBJECTS = [path.name for path in EEG_DIR.iterdir() if re.match(r'S\d*', path.name)]
+# Define a target directory for epoched data and make sure the directory is created
+EPOCH_DIR = DATA_ROOT / 'Epochs'
+EPOCH_DIR.mkdir(exist_ok=True)
+
+tstart = -0.1
+tstop = 1
+
+# +
+# Load stimuli
+# ------------
+# Load the gammatone-spectrograms; use the time axis of these as reference
+gammatone = [eelbrain.load.unpickle(PREDICTOR_DIR / f'{stimulus}~gammatone-8.pickle') for stimulus in STIMULI]
+# Resample the spectrograms to 100 Hz (time-step = 0.01 s), which we will use for TRFs
+gammatone = [x.bin(0.01, dim='time', label='start') for x in gammatone]
+# Pad onset with 100 ms and offset with 1 second; make sure to give the predictor a unique name as that will make it easier to identify the TRF later
+gammatone = [trftools.pad(x, tstart=-0.100, tstop=x.time.tstop + 1, name='gammatone') for x in gammatone]
+
+# load word onsets - gammatone used as time axis reference
+word_tables = [eelbrain.load.unpickle(PREDICTOR_DIR / f'{stimulus}~word.pickle') for stimulus in STIMULI]
```

Comment:
Don't these tables already contain the word onset times? It should not be necessary to load all these other predictors and turn word onsets into a uniform time series and back?